File: projetoJava/src/Carro.java
public class Carro {
    private String marca;
    private String modelo;
    private String cor;
    private Motor motor;
    private int velocidade;

    public Carro(String marca, String modelo, String cor, Motor motor) {
        this.marca = marca;
        this.modelo = modelo;
        this.cor = cor;
        this.motor = motor;
        this.velocidade = 0;
    }

    public void acelerar(int incremento) {
        this.velocidade += incremento;
        System.out.println("O carro acelerou para " + velocidade + " km/h.");
    }

    public void frear(int decremento) {
        this.velocidade -= decremento;
        if (this.velocidade < 0) {
            this.velocidade = 0;
        }
        System.out.println("O carro reduziu a velocidade para " + velocidade + " km/h.");
    }

    public void exibirDetalhes() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Cor: " + cor + ", Motor: " + motor.exibeMotor());
    }

    public void exibirVelocidade() {
        System.out.println("A velocidade atual do " + marca + " " + modelo + " é " + velocidade + " km/h.");
    }
}

File: projetoGo/carro.go
package main

import "fmt"

type Carro struct {
	Marca      string
	Modelo     string
	Cor        string
	Motor      Motor
	Velocidade int
}

func (c *Carro) Acelerar(incremento int) {
	c.Velocidade += incremento
	fmt.Printf("O carro acelerou para %d km/h.\n", c.Velocidade)
}

func (c *Carro) Frear(decremento int) {
	c.Velocidade -= decremento
	if c.Velocidade < 0 {
		c.Velocidade = 0
	}
	fmt.Printf("O carro reduziu a velocidade para %d km/h.\n", c.Velocidade)
}

func (c Carro) ExibirVelocidade() {
	fmt.Printf("A velocidade atual do %s %s é %d km/h.\n", c.Marca, c.Modelo, c.Velocidade)
}

func (c Carro) ExibirDetalhes() {
	fmt.Printf("Marca: %s, Modelo: %s, Cor: %s, Motor: %s\n", c.Marca, c.Modelo, c.Cor, c.Motor.ExibeMotor())
}
